src/main/java/com/android/volley/toolbox/DiskBasedAsyncCache.java

```diff
@@ -70,45 +75,53 @@ public void get(String key, final OnGetCompleteCallback callback) {
             final AsynchronousFileChannel afc =
                     AsynchronousFileChannel.open(path, StandardOpenOption.READ);
             channel = afc;
-            int headerSize = entry.getHeaderSize();
-            final int size = (int) file.length() - headerSize;
+            final int size = (int) file.length();
             final ByteBuffer buffer = ByteBuffer.allocate(size);
             afc.read(
                     /* destination= */ buffer,
-                    /* position= */ headerSize,
+                    /* position= */ 0,
                     /* attachment= */ null,
                     new CompletionHandler<Integer, Void>() {
                         @Override
                         public void completed(Integer result, Void v) {
                             closeChannel(afc, "completed read");
-                            // if the file size changes, return null
+                            CacheHeader entryOnDisk = CacheHeader.readHeader(buffer);
                             if (size != result) {
```

Comment:
yeah, that's what I was struggling with, since they all are slightly different. Would the best default be to remove as well as delete for all three? Or not delete in any case, and let pruning take care of that?

Reply:
my inclination is to fully prune an entry that we're unable to read regardless of the specific error that occurred.

Reply:
okay, that makes sense. Should we still log the specific issues in each case? Or should we group them all together, and just log that the file couldn't be read?

Reply:
Ideally we'd log the specific issue as that might help diagnose the nature in which some corruption was happening (or perhaps a bug in one of our checks).

Reply:
okay, I log the error then call a helper that removes the key, deletes the file, and calls the callback in each of these three cases. Should that be true for if the read fails (goes into failed function of completion handler)?

Reply:
Seems the same as any other kind of IO failure to me :)